src/adam7.rs

```diff
@@ -126,6 +200,39 @@ impl Iterator for Adam7Iterator {
     }
 }
 
+/// The algorithm to use when progressively filling pixel data from Adam7 interlaced passes.
+///
+/// Adam7 interlacing is a technique optionally used in PNG by which only a sub-sample of pixel
+/// data is encoded in the beginning of the image data chunks, followed by progressively larger
+/// subsets of the data in subsequent passes. Therefore a 'rough image' is available after ust a
+/// very tiny fraction of the data has been read which can be advantageous for loading an image
+/// from a slow IO medium while optimizing time-to-first-meaningful-paint and then replacing the
+/// presented data as it is streamed in.
+///
+/// There are trade-offs to make here. The strictly necessary requirement for an implementation is
+/// that the exact image is recovered after all passes are applied. However the intermediate states
+/// of the output are left to the implementation, as long as it follows the restriction of
+/// resulting in the intended image when all passes have been applied.
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
+#[non_exhaustive]
+pub enum Adam7Variant {
```

Comment:
While no longer strictly necessary in the API this is a nice documentation entry point. We can also make future use of this type when either the decoder is changed or we provide a wrapper around an image buffer (e.g. support for strided output buffers would be nice and somewhat painful).